php/media/class-gallery.php

```diff
@@ -0,0 +1,323 @@
+			)
+		);
+	}
+
+	/**
+	 * Check if WooCommerce is active.
+	 *
+	 * @return bool
+	 */
+	protected function woocommerce_active() {
+		return class_exists( 'WooCommerce' );
+	}
+
+	/**
+	 * Checks if the Cloudinary Gallery Widget is enabled.
+	 *
+	 * @return bool
+	 */
+	public function gallery_enabled() {
+		return isset( $this->original_config['enable_gallery'] ) && 'on' === $this->original_config['enable_gallery'];
+	}
+
+	/**
+	 * Fetches image public id and transformations.
+	 *
+	 * @param array|int[]|array[] $images An array of image IDs or a multi-dimensional array with url and id keys.
+	 *
+	 * @return array
+	 */
+	public function get_image_data( array $images ) {
+		$image_data = array();
+
+		foreach ( $images as $index => $image ) {
+			$image_id  = is_int( $image ) ? $image : $image['id'];
+			$image_url = is_int( $image ) ? $this->media->cloudinary_url( $image_id ) : $image['url'];
+
+			$image_data[ $index ]             = array();
+			$image_data[ $index ]['publicId'] = $this->media->get_public_id( $image_id );
+
+			$transformations = $this->media->get_transformations_from_string( $image_url );
+
+			if ( $transformations ) {
+				$image_data[ $index ]['transformation'] = array( 'transformation' => $transformations );
+			}
+		}
+
+		return $image_data;
+	}
+
+	/**
+	 * This rest endpoint handler will fetch the public_id and transformations off of a list of images.
+	 *
+	 * @param \WP_REST_Request $request The request.
+	 *
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function rest_cloudinary_image_data( \WP_REST_Request $request ) {
+		$request_body = json_decode( $request->get_body(), true );
+
+		if ( empty( $request_body ) || empty( $request_body['images'] ) ) {
+			return new \WP_Error( 400, 'The "images" key must be present in the request body.' );
+		}
+
+		$image_data = $this->get_image_data( $request_body['images'] );
+
+		return new \WP_REST_Response( $image_data );
+	}
+
+	/**
+	 * Add endpoints to the \Cloudinary\REST_API::$endpoints array.
+	 *
+	 * @param array $endpoints Endpoints from the filter.
+	 *
+	 * @return array
+	 */
+	public function rest_endpoints( $endpoints ) {
+
+		$endpoints['image_data'] = array(
+			'method'              => \WP_REST_Server::CREATABLE,
+			'callback'            => array( $this, 'rest_cloudinary_image_data' ),
+			'args'                => array(),
+			'permission_callback' => function() {
+				return current_user_can( 'edit_posts' );
+			},
+		);
+
+		return $endpoints;
+	}
+
+	/**
+	 * Setup hooks for the gallery.
+	 */
+	public function setup_hooks() {
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_gallery_library' ) );
+		add_filter( 'cloudinary_api_rest_endpoints', array( $this, 'rest_endpoints' ) );
+		add_action( 'enqueue_block_editor_assets', array( $this, 'block_editor_scripts_styles' ) );
+
+		if ( ! is_admin() ) {
+			// phpcs:disable
+			add_filter( 'woocommerce_single_product_image_thumbnail_html', function () { return ''; } );
```

Comment:
Can we use instead ['__return_empty_string'](https://developer.wordpress.org/reference/functions/__return_empty_string/)?

Reply:
Thanks!